src/include/block.h

```diff
@@ -308,6 +308,11 @@ struct __wt_block {
 
     /* Multi-file support */
     uint32_t read_count; /* Count of active read requests using this block handle */
+
+/* AUTOMATIC FLAG VALUE GENERATION START 0 */
+#define WT_BLOCK_TIERED_OBJECT_BACKING 0x1u
```

Comment:
As suggested in a separate comment, it might be more transparent to make this flag something like `WT_BLOCK_READONLY`. 

There are several other `bool` fields scattered around the `WT_BLOCK`. So it seems awkward to add a flag word with only one flag value.

It think it would be OK to just add another bool and create a ticket (not for this project) to consolidate things into a flag word. This could be a fine neweng/intern ticket. Or you could do that refactoring here.

Reply:
Fair enough - I've done the larger refactoring as part of this PR. The pitfall here is concurrently modifying unrelated flags - my understanding is concurrency isn't a concern once we're in the block manager, but I'd like to check that. If there's any doubt I'll revert my last commit and just add read-only as a flag.

Reply:
Ack! Sorry, I wasn't thinking about the concurrency angle. The block manger is definitely *not* single threaded. We can have multiple read/write requests in flight at once. But I expect many of those flags are used in places where we are single threaded -- for example verify should be a single thread with exclusive access to the Btree.

But I think it was be safer to revert this and consider it a bit more. 

I'd like to say that I wanted to make a point that you shouldn't always assume that senior folks know what they're talking about, but I'm afraid I'm not that clever.

Reply:
No worries - I didn't sink a heap of time into the conversion. The code should be fixed as of 683f1ee, with another standalone bool.